### ISSUE TITLE
How to understand two stages of training?

### ISSUE THREAD

**Issue by Martin-top** [hobincar/RecNet#8]
Hi, I'm a beginner. I have some questions I'd like to ask you.
I see that the code you provided is divided into two stages for training. I want to know what are the two stages respectively?Can you describe it briefly?
I understand these two phases as "Encoder-Decoder" and "Reconstructor", right?

Looking forward to your reply. Thank you very much.

**Comment by hobincar** [comment]
Hi,

As described in the [paper](http://openaccess.thecvf.com/content_cvpr_2018/papers/Wang_Reconstruction_Network_for_CVPR_2018_paper.pdf), the training is divided into two sequential steps; (1) training encoder-decoder, and (2) training encoder-decoder-reconstructor.

For the first step of training, I trained the encoder-decoder by simply setting the loss weight of reconstructor to zero in `config_stage1.py`. Then, starting from the weights of pre-trained encoder-decoder, I trained the full encoder-decoder-reconstructor by setting the loss weight of reconstructor as described in the [paper](http://openaccess.thecvf.com/content_cvpr_2018/papers/Wang_Reconstruction_Network_for_CVPR_2018_paper.pdf) (0.2 for global and 0.1 for local).

Therefore, the differences between the two config files are the loss weight of reconstructor, and a few hyperparameters that I found myself (e.g. epochs, learning rate, and when to start decaying learning rate).
